azure-functions-java-sdktypes/src/main/java/com/microsoft/azure/functions/sdktype/SdkTypeRegistry.java

```diff
@@ -0,0 +1,49 @@
+package com.microsoft.azure.functions.sdktype;
+
+import com.microsoft.azure.functions.sdktype.blob.BlobClientSdkTypeFactory;
+import com.microsoft.azure.functions.sdktype.blob.BlobContainerSdkTypeFactory;
+
+import java.lang.reflect.Parameter;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A registry that knows about recognized SDK client FQCNs and can create SdkType objects.
+ */
+public class SdkTypeRegistry {
+    // Maps FQCN -> Class<? extends SdkType>
+    private final Map<String, SdkTypeFactory> knownTypes = new HashMap<>();
+
+    public SdkTypeRegistry() {
+        knownTypes.put("com.azure.storage.blob.BlobClient", new BlobClientSdkTypeFactory());
+        knownTypes.put("com.azure.storage.blob.BlobContainerClient", new BlobContainerSdkTypeFactory());
+    }
+
+    /** Check if we recognize a param type */
+    public boolean isRecognizedType(String fqcn) {
+        return knownTypes.containsKey(fqcn);
+    }
+
+    /**
+     * Build-time usage: create a minimal SdkTypeMetaData from the recognized fqcn + param.
+     */
+    public SdkTypeMetaData createMetaData(String fqcn, Parameter param) throws Exception {
+        SdkTypeFactory factory = knownTypes.get(fqcn);
+        if (factory == null) {
```

Comment:
If you use @NotNull these embedded code could be replace and code would be cleaner